docs/source/model-registry.rst

```diff
@@ -78,47 +79,170 @@ version to one of the other valid stages.
+
+    {   'creation_timestamp': 1582671933216,
+        'description': None,
+        'last_updated_timestamp': 1582671960712,
+        'latest_versions': [<ModelVersion: creation_timestamp=1582671933246, current_stage='Production', description='A random forest model containing 100 decision trees trained in scikit-learn', last_updated_timestamp=1582671960712, name='sk-learn-random-forest-reg-model', run_id='ae2cc01346de45f79a44a320aab1797b', source='./mlruns/0/ae2cc01346de45f79a44a320aab1797b/artifacts/sklearn-model', status='READY', status_message=None, user_id=None, version=1>,
+                           <ModelVersion: creation_timestamp=1582671960628, current_stage='None', description=None, last_updated_timestamp=1582671960628, name='sk-learn-random-forest-reg-model', run_id='d994f18d09c64c148e62a785052e6723', source='./mlruns/0/d994f18d09c64c148e62a785052e6723/artifacts/sklearn-model', status='READY', status_message=None, user_id=None, version=2>],
+        'name': 'sk-learn-random-forest-reg-model'}
+    ...
+    ...
+
+With hundreds of models, it can be cumbersome to peruse the results returned from this call. A more efficient approach would be to search for a specific model name and list its version
+details using `search_model_versions(...) <https://www.mlflow.org/docs/latest/python_api/mlflow.tracking.html#mlflow.tracking.MlflowClient.search_model_versions>`_ method
+and provide a filter string such as ``“name=’sk-learn-random-forest-reg-model’”``
+
+.. code-block:: py
 
-You can query for the latest version of each registered model by stage using the ``get_latest_versions`` API:
+    client=MlflowClient()
+    [pprint.pprint(dict(mv), indent=4) for mv in client.search_model_versions(“name=’sk-learn-random-forest-reg-model’”)]
+
+    {   'creation_timestamp': 1582671933246,
+        'current_stage': 'Production',
+        'description': 'A random forest model containing 100 decision trees '
+                       'trained in scikit-learn',
+        'last_updated_timestamp': 1582671960712,
+        'name': 'sk-learn-random-forest-reg-model',
+        'run_id': 'ae2cc01346de45f79a44a320aab1797b',
+        'source': './mlruns/0/ae2cc01346de45f79a44a320aab1797b/artifacts/sklearn-model',
+        'status': 'READY',
+        'status_message': None,
+        'user_id': None,
+        'version': 1}
+
+    {   'creation_timestamp': 1582671960628,
+        'current_stage': 'None',
+        'description': None,
+        'last_updated_timestamp': 1582671960628,
+        'name': 'sk-learn-random-forest-reg-model',
+        'run_id': 'd994f18d09c64c148e62a785052e6723',
+        'source': './mlruns/0/d994f18d09c64c148e62a785052e6723/artifacts/sklearn-model',
+        'status': 'READY',
+        'status_message': None,
+        'user_id': None,
+        'version': 2
+    }
+
+Deleting Registered Models
+--------------------------
+You can either delete a specific version of the registered model or registered model and all its versions.
```

Comment:
Clause is a bit confusing, maybe repeat "can delete":

`You can either delete a specific version of the registered model or you can delete the registered model and all its versions.`

Reply:
Will fix it.

Reply:
"Archived" is a stage for a model version is a part of lifecycle stage. Usually people would want to keep archived models around for long. Deleting as you mentioned is permanently deleting the model version from the registry. I am not super psyched about mixing these 2 concepts together. How about this.

```
Archiving Models
----------------

To transition model versions out of **Production** stage, transition those model into **Archived** 
stage.

.. code-block:: py

    client = MlflowClient()
    client.transition_model_version_stage(name="sk-learn-random-forest-reg-model",
  	        version=3,
	          stage="Archived")


Deleting Models
---------------

.. note::
  Deleting registered models and model versions is irrevocable, so use it judiciously.

You can either delete a specific version of a registered model or you can delete a registered model 
and all its versions.

.. code-block:: py

    # Delete models versions 1, 2, and 3
    versions = [1, 2, 3]
    client=MlflowClient()
    for version in versions:
      client.delete_model_version(name="sk-learn-random-forest-reg-model", version=version)

    # Delete the registered model along with all its versions
    client.delete_registered_model(name="sk-learn-random-forest-reg-model")

```